meta-mbl-distro/scripts/create-update-payload

```diff
@@ -206,31 +284,26 @@ def get_path_for_bootloader_slot(bootloader_slot_name, deploy_dir, tinfoil):
     """
     filename_var_name = "MBL_{}_FILENAME".format(bootloader_slot_name)
     filename = get_bitbake_conf_var(filename_var_name, tinfoil)
-    if not filename:
-        bb.fatal("Failed to get value of {}".format(filename_var_name))
-    return deploy_dir / filename
+    return ArchivedFileSpec(
+        deploy_dir / filename, "{}.xz".format(bootloader_slot_name)
+    )
 
 
-def get_files_for_boot_part(deploy_dir, tinfoil=None):
+def get_archived_file_specs_for_boot_part(deploy_dir, tinfoil=None):
     """
-    Match all files for the BOOT partition (or BLFS).
-
-    Looks up the value of the IMAGE_BOOT_FILES bb variable. Returns a list of
-    paths as the value.
+    Get ArchivedFileSpecs for the boot partition (or BLFS).
 
     :param deploy_dir Path: path to DEPLOY_DIR_IMAGE.
     :param tinfoil Tinfoil: BitBake tinfoil object.
     """
-    image_boot_files = get_bitbake_conf_var("IMAGE_BOOT_FILES", tinfoil)
-    if not image_boot_files:
-        bb.fatal("Failed to find any files in IMAGE_BOOT_FILES")
-
-    boot_file_specs = image_boot_files.split()
+    boot_file_entries = get_bitbake_conf_var(
+        "IMAGE_BOOT_FILES", tinfoil
+    ).split()
     boot_files = list(
         itertools.chain(
             *[
-                img_boot_files_val_to_map_val(elem, deploy_dir)
-                for elem in boot_file_specs
+                img_boot_files_val_to_archived_file_specs(elem, deploy_dir)
```

Comment:
I know I already approved this, but I'm wondering if we can do something like 
`list(itertools.chain.from_iterable(img_boot_files_val_to_archived_file_specs(elem, deploy_dir) for elem in boot_file_entries))` and save building the extra list in the comprehension?

Reply:
Yeah, that is a bit neater.

Reply:
https://github.com/ARMmbed/meta-mbl/pull/730/commits/eacb96ea2f1ec15419d9f415969e7058940bc456